### PR TITLE
Refactoring backend acording to the UI change

### PR DESCRIPTION
**Pull request by DMHP**

### CODE REVIEW

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java [hunk 1]
```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
      *
      * @return auto-generated one time password
      */
-    public String generatePassode() {
-        return IdentityUtils.generatePasscode(6);
+    public String generatePasscode() {
+        return IdentityUtils.getInstance().generatePasscode(6);
```
Comment (GayanM) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java]: define default value in constant file

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java [hunk 2]
```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
...
      *
      * @param userUniqueId selected user id
-     * @param passCode     generated code
      * @throws IdentityRecoveryException
      */
-    public void persistPasscode(String userUniqueId, String passCode) throws IdentityRecoveryException {
+    public String persistPasscode(String userUniqueId) throws IdentityRecoveryException {
         UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
-        UserRecoveryData recoveryDataDO = new UserRecoveryData(userUniqueId, passCode,
+        String passcode = generatePasscode();
```
Comment (GayanM) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java]: can invoke util method directly

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java [hunk 3]
```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
      *
      * @return auto-generated one time password
      */
-    public String generatePassode() {
-        return IdentityUtils.generatePasscode(6);
```
Comment (GDRDABARERA) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java]: This value "6" should come from a config file in mappings

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java [hunk 4]
```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
...
-     * @param passCode     generated code
      * @throws IdentityRecoveryException
      */
-    public void persistPasscode(String userUniqueId, String passCode) throws IdentityRecoveryException {
+    public String persistPasscode(String userUniqueId) throws IdentityRecoveryException {
         UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
-        UserRecoveryData recoveryDataDO = new UserRecoveryData(userUniqueId, passCode,
+        String passcode = generatePasscode();
+        if (log.isDebugEnabled()) {
+            log.debug("The passcode is generated.");
```
Comment (Pushpalanka) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java]: Add more details to the log. May be we can print the userid and say passcode is persisted for this x user.

File: components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java [hunk 5]
```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
...
+        }
+        UserRecoveryData recoveryDataDO = new UserRecoveryData(userUniqueId, passcode,
                 RecoveryScenarios.ADMIN_FORCED_PASSWORD_RESET_VIA_OTP, RecoverySteps.UPDATE_PASSWORD);
-        userRecoveryDataStore.invalidateUserScenario(userUniqueId, "ADMIN_FORCED_PASSWORD_RESET_VIA_OTP");
+        userRecoveryDataStore.invalidateUserScenario(userUniqueId,
+                String.valueOf(RecoveryScenarios.ADMIN_FORCED_PASSWORD_RESET_VIA_OTP));
         userRecoveryDataStore.store(recoveryDataDO);
+
+        if (log.isDebugEnabled()) {
+            log.debug("The passcode is persisted.");
```
Comment (Pushpalanka) [inline components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java]: Add more details to the log.

**Comment by DMHP** [comment]
https://github.com/wso2/carbon-identity-mgt/pull/180

**Pull request closed without merging**